ddtrace/internal/runtime/runtime_metrics.py

```diff
@@ -0,0 +1,112 @@
+import threading
+import time
+import itertools
+
+from ..logger import get_logger
+from .constants import (
+    DEFAULT_RUNTIME_METRICS,
+    DEFAULT_RUNTIME_TAGS,
+)
+from .metric_collectors import (
+    GCRuntimeMetricCollector,
+    PSUtilRuntimeMetricCollector,
+)
+from .tag_collectors import (
+    TracerTagCollector,
+    PlatformTagCollector,
+)
+
+log = get_logger(__name__)
+
+
+class RuntimeCollectorsIterable:
+    def __init__(self, enabled=None):
+        self._enabled = enabled or self.ENABLED
+        # Initialize the collectors.
+        self._collectors = [c() for c in self.COLLECTORS]
+
+    def __iter__(self):
+        collected = (
+            collector.collect(self._enabled)
+            for collector in self._collectors
+        )
+        return itertools.chain.from_iterable(collected)
+
+    def __repr__(self):
+        return '{}(enabled={})'.format(
+            self.__class__,
+            self._enabled,
+        )
+
+
+class RuntimeTags(RuntimeCollectorsIterable):
+    ENABLED = DEFAULT_RUNTIME_TAGS
+    COLLECTORS = [
+        TracerTagCollector,
+        PlatformTagCollector,
+    ]
+
+
+class RuntimeMetrics(RuntimeCollectorsIterable):
+    ENABLED = DEFAULT_RUNTIME_METRICS
+    COLLECTORS = [
+        GCRuntimeMetricCollector,
+        PSUtilRuntimeMetricCollector,
+    ]
+
+
+class RuntimeWorker(object):
+    FLUSH_INTERVAL = 10
+
+    def __init__(self, statsd_client, flush_interval=None):
+        self._lock = threading.Lock()
+        self._stay_alive = None
+        self._thread = None
+        self._flush_interval = flush_interval or self.FLUSH_INTERVAL
+        self._statsd_client = statsd_client
+        self._runtime_metrics = RuntimeMetrics()
+
+    def _target(self):
+        while True:
```

Comment:
`while self._stay_alive` sounds simpler here.

There's no need for any lock. Assignation/variable reading is always atomic in Python.